README.md
# Overview

Repository and Issue Tracker can be found at https://github.com/evoWeb/sf_register

Suits all your needs to handle frontend users like register new users, edit data and change password.

So whats already in there?

- creating frontend user
    - send notification to user and admin
    - activate via link in email by user or admin
    - notification email after activation
    - configure email addresses for user and admin mails separately
    - set different usergroup pre and post activation
    - general terms and conditions acception as checkbox
    - old password verification before setting new password
    - password strength indicator without need of any js lib
    - email/password repeat validation
    - profilimage upload, remove and edit with plain or encrypted filename
    - country as selectbox (values from static_info_tables)
    - country zone as selectbox (values from static_info_tables)
    - country zone change with ajax if country selectbox changed
    - language as selectbox (values from static_info_tables)
    - gender as radiobox
    - title as textbox and selectbox
    - pseudonym
    - timezone as selectbox
    - daylight saving as checkbox
    - privacy agreement as checkbox
    - salutation as radiobuttons and selectbox
    - birthdate as selectboxes
    - captcha with integration of existing captcha extensions
    - configuration email as username
- custom validators
    - user model
    - captcha
    - required
    - repeat
- custom viewhelpers
    - required
    - captcha
    - static info tables selectboxes
- edit frontend user
- change password
- different template file for every form, preview, save and email view, configurable so they do not need to stay in extension
- override template rootpath in plugin
- realurl settings preconfigured for controller, actions and mailhash

If all that is already in, what is missing?

- complete documentation
- ajax handling
    - javascript validators in jquery, extjs you name it
- add interface for user model to enable other extension to extend the model (still needs changes to extbase)
- better extendability of frontend user model, well this needs some love in extbase
- multistep creation and editing

How could you help?

- file issues about bugs and if you already have a solution send the patch in
- sponsor features you are in need of

Homepage http://www.evoweb.de/


ChangeLog
2012-01-06 10:52  sfischer

	* ext_emconf.php: Lower requirements for branch 1.4.0

2012-01-02 08:43  sfischer

	* Classes/Utility/WizardIcon.php: Fix translation handling in
	  wizard icon

2011-09-18 13:11  sfischer

	* .: Create branch for 4.5 from trunk

2011-09-18 12:39  sfischer

	* Add set activated on to 1.1.1970 to ensure activation is possible

2011-09-18 12:36  sfischer

	* Change german translation

2011-09-18 12:32  sfischer

	* Change labels for the confirmation step

2011-08-31 13:23  sfischer

	* Fix usage of language for wizard

2011-07-24 14:56  sfischer

	* Raise extbase and fluid requirements

2011-07-19 07:50  sfischer

	* Bugfix hook

2011-07-19 07:38  sfischer

	* Change textbox to textfield as textbox is deprecated in 4.6

2011-07-19 07:37  sfischer

	* Raise version and requirement to prevent extension gets loaded on
	  systems that lead to errors

2011-07-19 07:36  sfischer

	* Add save hook to be able to manipulate the user object before it
	  gets added

2011-07-19 07:35  sfischer

	* Add save hook to be able to manipulate the user object before it
	  gets updated

2011-07-19 06:48  sfischer

	* Add save hook to password controller

2011-07-19 06:45  sfischer

	* Make hook service extendable for other extensions

2011-07-19 06:43  sfischer

	* Add advanced hooking service

2011-07-16 08:27  sfischer

	* Set sessionKey to sf_register

2011-07-16 08:16  sfischer

	* Remove call of store

2011-07-16 08:16  sfischer

	* Add auto store to remove of values

2011-07-16 07:59  sfischer

	* Auto store on destruct

2011-07-16 07:38  sfischer

	* Check if captchaWasValidPreviously was set to true previously
	  If Check failed the captcha gets validated and the result is
	  stored in session

2011-07-16 07:29  sfischer

	* Unset captchaWasValidPreviously on render of captcha

2011-07-16 07:28  sfischer

	* Format code

2011-07-16 07:27  sfischer

	* Unset captchaWasValidPreviously from session on store

2011-07-16 06:40  sfischer

	* Add session service

2011-07-12 11:58  sfischer

	* Add content element wizard icon
	  Thanks go to Jonas Dübi for handing an patch as base for this
	  feature

2011-07-09 18:14  sfischer

	* Fixes #28047 check for user model before prepareDateOfBirth gets
	  called

2011-07-09 18:00  sfischer

	* fix whitespaces

2011-07-09 15:56  sfischer

	* Fix missing returns
	  Fixing missing $options

2011-07-09 15:48  sfischer

	* Fix php doc headers

2011-07-09 15:41  sfischer

	* Replace password repeat with password again

2011-07-09 15:20  sfischer

	* Add semicolon to lines
	  Add var to variable definition

2011-07-09 15:18  sfischer

	* Remove unused fixture

2011-07-09 15:02  sfischer

	* Remove usage of deprecated methods

2011-07-09 14:43  sfischer

	* Change from get object to create object

2011-07-09 12:56  sfischer

	* Correct phpdoc headers
	  Remove comment

2011-07-08 21:32  sfischer

	* Add method to send emails
	  Add boolean check if notification or activation email should be
	  send to user or admin

2011-07-08 21:30  sfischer

	* Add methods to send emails to user or admin with notification or
	  activation message

2011-07-08 21:28  sfischer

	* Add settings to enable email notifications after editing

2011-07-08 21:25  sfischer

	* Add labels for typoscript constants

2011-07-08 20:50  sfischer

	* Add templates for emails after edit to notify or activate

2011-07-08 19:32  sfischer

	* Fix existing unit tests after a lot of changes in extension

2011-07-07 08:50  sfischer

	* Change validation call for actions

2011-07-07 08:46  sfischer

	* Remove linebreak

2011-07-07 08:46  sfischer

	* Harding the sql query
	  Shorten getter return

2011-07-07 08:45  sfischer

	* Change instanciating from concrete model to interface

2011-07-06 08:50  sfischer

	* Move setModel and setFieldname to getValidator method

2011-07-06 08:38  sfischer

	* Fix a template error after renaming of passwordAgain to
	  passwordRepeat

2011-07-06 08:17  sfischer

	* Cleanup and documentation

2011-07-05 15:15  sfischer

	* Add comments to classes

2011-07-05 15:12  sfischer

	* Replace usage of deprecated methods

2011-07-05 14:42  sfischer

	* Display errors if the captcha is not installed but choosen

2011-07-05 14:36  sfischer

	* Fix fatal error on validate if jm_recaptcha is choosen but not
	  installed

2011-07-05 14:35  sfischer

	* Fix fatal error on submit if sr_freecap is choosen as captcha but
	  is not installed

2011-07-05 14:26  sfischer

	* Resolve #27859 Remove miss pasted line

2011-07-05 11:55  sfischer

	* Resolves: #27853 Change to only add settings if realurl
	  configuration is present

2011-07-05 11:50  sfischer

	* Fixing load of realurl configuration from ext_tables to
	  ext_localconf so it gets included every page load

2011-07-05 10:37  sfischer

	* Fixing include of realurl setting even though realurl is not
	  available

2011-07-05 10:32  sfischer

	* Cleanup and documentation

2011-07-05 09:38  sfischer

	* Cleanup and documentation

2011-07-05 09:38  sfischer

	* Cleanup and documentation

2011-07-05 09:36  sfischer

	* Cleanup and documentation

2011-07-05 09:35  sfischer

	* Cleanup and documentation

2011-07-05 09:35  sfischer

	* Cleanup and a lot of documentation in PHPdoc headers

2011-06-30 06:46  sfischer

	* Remove type from uservalidation call because of previous commit

2011-06-30 06:43  sfischer

	* Decouple uservalidator from options and take framework
	  configurations instead to determin what rules to validate with

2011-06-28 06:55  sfischer

	* Cleanup GPL phpdoc headers

2011-06-25 11:40  sfischer

	* Add getter and setter for dateOfBirth[Day|Month|Year] to be able
	  to select the date of birth with selectboxes

2011-06-25 07:41  sfischer

	* Change to activation process to keep unknown, already activated
	  and now active state apart

2011-06-24 09:36  sfischer

	* Add eid for ajax requests
	  Add country zone replacement with ajax on country change

2011-06-23 07:10  sfischer


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "sf_register".
#
# Auto generated 13-04-2012 14:54
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'FeUser Register',
	'description' => 'Offers the possibility to maintain the fe_user data in frontend by the user self.',
	'category' => 'Sebastian Fischer',
	'shy' => 0,
	'version' => '1.4.0',
	'dependencies' => 'extbase,fluid',
	'conflicts' => '',
	'priority' => 'bottom',
	'loadOrder' => '',
	'module' => '',
	'state' => 'beta',
	'uploadfolder' => 0,
	'createDirs' => 'typo3temp/sf_register',
	'modify_tables' => 'fe_users',
	'clearcacheonload' => 1,
	'lockType' => '',
	'author' => 'Sebastian Fischer',
	'author_email' => 'typo3@evoweb.de',
	'author_company' => 'evoweb',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'php' => '5.3.0-0.0.0',
			'typo3' => '4.5.0-0.0.0',
			'extbase' => '1.3.0-0.0.0',
			'fluid' => '1.3.0-0.0.0',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'suggests' => array(
	),
	'_md5_values_when_last_written' => 'a:105:{s:9:"ChangeLog";s:4:"736c";s:21:"ext_conf_template.txt";s:4:"1623";s:12:"ext_icon.gif";s:4:"c675";s:17:"ext_localconf.php";s:4:"65dc";s:14:"ext_tables.php";s:4:"45f8";s:14:"ext_tables.sql";s:4:"2100";s:20:"Classes/Api/Ajax.php";s:4:"6a61";s:39:"Classes/Controller/FeuserController.php";s:4:"6ae9";s:45:"Classes/Controller/FeuserCreateController.php";s:4:"d79b";s:43:"Classes/Controller/FeuserEditController.php";s:4:"6bfe";s:47:"Classes/Controller/FeuserPasswordController.php";s:4:"04ec";s:37:"Classes/Domain/Model/FrontendUser.php";s:4:"0251";s:33:"Classes/Domain/Model/Password.php";s:4:"9e15";s:38:"Classes/Domain/Model/StaticCountry.php";s:4:"bf58";s:42:"Classes/Domain/Model/StaticCountryZone.php";s:4:"62f4";s:39:"Classes/Domain/Model/StaticLanguage.php";s:4:"adc8";s:52:"Classes/Domain/Repository/FrontendUserRepository.php";s:4:"7e71";s:53:"Classes/Domain/Repository/StaticCountryRepository.php";s:4:"86bf";s:57:"Classes/Domain/Repository/StaticCountryZoneRepository.php";s:4:"86e6";s:54:"Classes/Domain/Repository/StaticLanguageRepository.php";s:4:"d38b";s:45:"Classes/Domain/Validator/BadWordValidator.php";s:4:"f617";s:45:"Classes/Domain/Validator/CaptchaValidator.php";s:4:"8662";s:43:"Classes/Domain/Validator/EmptyValidator.php";s:4:"2403";s:58:"Classes/Domain/Validator/EqualCurrentPasswordValidator.php";s:4:"17f9";s:54:"Classes/Domain/Validator/EqualCurrentUserValidator.php";s:4:"898d";s:49:"Classes/Domain/Validator/ImageUploadValidator.php";s:4:"8706";s:44:"Classes/Domain/Validator/IsTrueValidator.php";s:4:"4939";s:44:"Classes/Domain/Validator/RepeatValidator.php";s:4:"7fba";s:46:"Classes/Domain/Validator/RequiredValidator.php";s:4:"ef10";s:44:"Classes/Domain/Validator/UniqueValidator.php";s:4:"19f2";s:42:"Classes/Domain/Validator/UserValidator.php";s:4:"ad78";s:30:"Classes/Interfaces/Captcha.php";s:4:"91db";s:35:"Classes/Interfaces/FrontendUser.php";s:4:"6cd3";s:25:"Classes/Services/File.php";s:4:"6f12";s:25:"Classes/Services/Hook.php";s:4:"90fe";s:26:"Classes/Services/Login.php";s:4:"57f9";s:25:"Classes/Services/Mail.php";s:4:"32cd";s:28:"Classes/Services/Session.php";s:4:"4190";s:44:"Classes/Services/Captcha/AbstractAdapter.php";s:4:"8dbe";s:50:"Classes/Services/Captcha/CaptchaAdapterFactory.php";s:4:"4344";s:47:"Classes/Services/Captcha/JmRecaptchaAdapter.php";s:4:"98a9";s:45:"Classes/Services/Captcha/SrFreecapAdapter.php";s:4:"f181";s:30:"Classes/Utility/WizardIcon.php";s:4:"3e13";s:40:"Classes/Validation/ValidatorResolver.php";s:4:"f3a0";s:46:"Classes/ViewHelpers/Form/CaptchaViewHelper.php";s:4:"6c51";s:50:"Classes/ViewHelpers/Form/RangeSelectViewHelper.php";s:4:"27ad";s:47:"Classes/ViewHelpers/Form/RequiredViewHelper.php";s:4:"22d5";s:60:"Classes/ViewHelpers/Form/SelectStaticCountriesViewHelper.php";s:4:"74de";s:63:"Classes/ViewHelpers/Form/SelectStaticCountryZonesViewHelper.php";s:4:"dc1a";s:59:"Classes/ViewHelpers/Form/SelectStaticLanguageViewHelper.php";s:4:"a145";s:51:"Classes/ViewHelpers/Form/SelectStaticViewHelper.php";s:4:"21c9";s:32:"Configuration/FlexForms/form.xml";s:4:"68f3";s:38:"Configuration/Realurl/realurl_conf.php";s:4:"ebe7";s:46:"Configuration/TypoScript/maximal/constants.txt";s:4:"111f";s:42:"Configuration/TypoScript/maximal/setup.txt";s:4:"7816";s:46:"Configuration/TypoScript/minimal/constants.txt";s:4:"1cb4";s:42:"Configuration/TypoScript/minimal/setup.txt";s:4:"c1ee";s:40:"Resources/Private/Language/locallang.xml";s:4:"a2a6";s:43:"Resources/Private/Language/locallang_be.xml";s:4:"0dd6";s:43:"Resources/Private/Language/locallang_ts.xml";s:4:"8e3c";s:50:"Resources/Private/Partials/captchaJmRecaptcha.html";s:4:"f60e";s:48:"Resources/Private/Partials/captchaSrFreecap.html";s:4:"33d2";s:42:"Resources/Private/Partials/fieldError.html";s:4:"3e25";s:42:"Resources/Private/Partials/formErrors.html";s:4:"c714";s:40:"Resources/Private/Partials/required.html";s:4:"73a8";s:46:"Resources/Private/Partials/selectTimezone.html";s:4:"80f7";s:67:"Resources/Private/Templates/Email/AdminActivationMailAfterEdit.html";s:4:"0fae";s:60:"Resources/Private/Templates/Email/AdminNotificationMail.html";s:4:"8893";s:69:"Resources/Private/Templates/Email/AdminNotificationMailAfterEdit.html";s:4:"8893";s:74:"Resources/Private/Templates/Email/AdminNotificationMailPostActivation.html";s:4:"d16b";s:73:"Resources/Private/Templates/Email/AdminNotificationMailPreActivation.html";s:4:"0fae";s:66:"Resources/Private/Templates/Email/UserActivationMailAfterEdit.html";s:4:"93f6";s:59:"Resources/Private/Templates/Email/UserNotificationMail.html";s:4:"0fc1";s:68:"Resources/Private/Templates/Email/UserNotificationMailAfterEdit.html";s:4:"0fc1";s:73:"Resources/Private/Templates/Email/UserNotificationMailPostActivation.html";s:4:"6adf";s:72:"Resources/Private/Templates/Email/UserNotificationMailPreActivation.html";s:4:"93f6";s:53:"Resources/Private/Templates/FeuserCreate/Confirm.html";s:4:"8ba0";s:50:"Resources/Private/Templates/FeuserCreate/Form.html";s:4:"f742";s:53:"Resources/Private/Templates/FeuserCreate/Preview.html";s:4:"eeef";s:50:"Resources/Private/Templates/FeuserCreate/Save.html";s:4:"ce5a";s:48:"Resources/Private/Templates/FeuserEdit/Form.html";s:4:"d6e2";s:51:"Resources/Private/Templates/FeuserEdit/Preview.html";s:4:"cea0";s:48:"Resources/Private/Templates/FeuserEdit/Save.html";s:4:"c9f3";s:52:"Resources/Private/Templates/FeuserPassword/Form.html";s:4:"6e37";s:52:"Resources/Private/Templates/FeuserPassword/Save.html";s:4:"cfb8";s:39:"Resources/Public/Images/progressbar.png";s:4:"af24";s:44:"Resources/Public/JavaScript/passwordmeter.js";s:4:"9635";s:42:"Resources/Public/JavaScript/sf_register.js";s:4:"da42";s:39:"Resources/Public/Stylesheets/styles.css";s:4:"a65a";s:41:"Tests/Controller/FeuserControllerTest.php";s:4:"5079";s:47:"Tests/Controller/FeuserCreateControllerTest.php";s:4:"8e5e";s:45:"Tests/Controller/FeuserEditControllerTest.php";s:4:"a960";s:49:"Tests/Controller/FeuserPasswordControllerTest.php";s:4:"8c35";s:39:"Tests/Domain/Model/FrontendUserTest.php";s:4:"0fdc";s:35:"Tests/Domain/Model/PasswordTest.php";s:4:"ffb5";s:54:"Tests/Domain/Repository/FrontendUserRepositoryTest.php";s:4:"98fb";s:47:"Tests/Domain/Validator/BadWordValidatorTest.php";s:4:"d008";s:60:"Tests/Domain/Validator/EqualCurrentPasswordValidatorTest.php";s:4:"5eb6";s:51:"Tests/Domain/Validator/ImageUploadValidatorTest.php";s:4:"3f24";s:46:"Tests/Domain/Validator/IsTrueValidatorTest.php";s:4:"5072";s:46:"Tests/Domain/Validator/UniqueValidatorTest.php";s:4:"0f1a";s:44:"Tests/Domain/Validator/UserValidatorTest.php";s:4:"cc4a";s:27:"Tests/Services/FileTest.php";s:4:"3965";s:27:"Tests/Services/MailTest.php";s:4:"4578";s:42:"Tests/Validation/ValidatorResolverTest.php";s:4:"992f";}',
);

?>